op-e2e/interop/supersystem.go

```diff
@@ -0,0 +1,622 @@
+	batcherFlags "github.com/ethereum-optimism/optimism/op-batcher/flags"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/devkeys"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/foundry"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/interopgen"
+	op_e2e "github.com/ethereum-optimism/optimism/op-e2e"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/fakebeacon"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/geth"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/opnode"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/services"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/setuputils"
+	"github.com/ethereum-optimism/optimism/op-node/node"
+	"github.com/ethereum-optimism/optimism/op-node/p2p"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/driver"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/sync"
+	l2os "github.com/ethereum-optimism/optimism/op-proposer/proposer"
+	"github.com/ethereum-optimism/optimism/op-service/client"
+	"github.com/ethereum-optimism/optimism/op-service/clock"
+	"github.com/ethereum-optimism/optimism/op-service/dial"
+	"github.com/ethereum-optimism/optimism/op-service/endpoint"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	"github.com/ethereum-optimism/optimism/op-service/metrics"
+	"github.com/ethereum-optimism/optimism/op-service/oppprof"
+	oprpc "github.com/ethereum-optimism/optimism/op-service/rpc"
+	"github.com/ethereum-optimism/optimism/op-service/sources"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	supervisorConfig "github.com/ethereum-optimism/optimism/op-supervisor/config"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor"
+)
+
+// SuperSystem is an interface for the system (collection of connected resources)
+// it provides a way to get the resources for a network by network ID
+// and provides a way to get the list of network IDs
+// this is useful for testing multiple network backends,
+// for example, interopE2ESystem is the default implementation, but a shim to
+// kurtosis or another testing framework could be implemented
+type SuperSystem interface {
+	// get the supervisor
+	Supervisor() *supervisor.SupervisorService
+	// get the supervisor client
+	SupervisorClient() *sources.SupervisorClient
+	// get the batcher for a network
+	Batcher(network string) *bss.BatcherService
+	// get the proposer for a network
+	Proposer(network string) *l2os.ProposerService
+	// get the opnode for a network
+	OpNode(network string) *opnode.Opnode
+	// get the geth instance for a network
+	L2Geth(network string) *geth.GethInstance
+	// get the L2 geth client for a network
+	L2GethClient(network string) *ethclient.Client
+	// get the secret for a network and role
+	L2OperatorKey(network string, role devkeys.ChainOperatorRole) ecdsa.PrivateKey
+	// get the list of network IDs
+	L2IDs() []string
```

Comment:
Any reason that this is `[]string` rather than `[]*big.Int`? Chain id type is generally `*big.Int`

Reply:
It's using strings because that's what the `L2s` map uses for a key, and the map uses strings because that's what it's getting from the World Output:

```
type WorldOutput struct {
	L1  *L1Output
	L2s map[string]*L2Output
}
```
The purpose behind calling `L2IDs` would be to have a list of keys you can use to reference the individual L2s, like so:

```
ids := sys.L2IDs()
batcher0 := sys.Batcher(ids[0])
```

All `l2Set` structures also contain their ChainID as a Big Int, so if it's needed you can always get it. But for keying into the system structure, Strings seemed most compatible with the output I'm processing.

If you feel strongly about making it all conform to big int I can work on that, but I think we'd want the World Output to do the same in that case (which is outside the scope of this PR)

Reply:
upon 30 more seconds of consideration, I am going to convert things to big int across SuperSystem. No need to modify world output.